bokeh/models/widgets/inputs.py

```diff
@@ -43,6 +43,10 @@ class TextInput(InputWidget):
     A callback to run in the browser whenever the user unfocuses the TextInput
     widget by hitting Enter or clicking outside of the text box area.
     """)
+    
+    placeholder = String(default="", help="""
+    Placeholder for empty input field
+    """)
```

Comment:
Or maybe am I misunderstanding placeholder? I thought was something like `"*"` that would obscure a password field when a user types, but maybe not?